hack/generator/pkg/astmodel/storage_conversion_factories.go

```diff
@@ -77,75 +73,66 @@ func createTypeConversion(
 	return nil, err
 }
 
-// assignPrimitiveTypeFromPrimitiveType will generate a direct assignment if both types have the
-// same underlying primitive type and are not optional
+// assignToOptionalType will generate a conversion where the destination is optional, if the
+// underlying type of the destination is compatible with the source.
 //
-// <destination> = <source>
+// <destination> = &<source>
 //
-func assignPrimitiveTypeFromPrimitiveType(
+func assignToOptionalType(
 	sourceEndpoint *StorageConversionEndpoint,
 	destinationEndpoint *StorageConversionEndpoint,
-	_ *StorageConversionContext) StorageTypeConversion {
-
-	// Require source to be non-optional
-	if _, srcOpt := AsOptionalType(sourceEndpoint.Type()); srcOpt {
-		return nil
-	}
-
-	// Require destination to be non-optional
-	if _, dstOpt := AsOptionalType(destinationEndpoint.Type()); dstOpt {
-		return nil
-	}
+	conversionContext *StorageConversionContext) StorageTypeConversion {
 
-	// Require source to be a primitive type
-	srcPrim, srcOk := AsPrimitiveType(sourceEndpoint.Type())
-	if !srcOk {
+	// Require destination to be optional
+	dstOpt, dstIsOpt := AsOptionalType(destinationEndpoint.Type())
+	if !dstIsOpt {
+		// Destination is not optional
 		return nil
 	}
 
-	// Require destination to be a primitive type
-	dstPrim, dstOk := AsPrimitiveType(destinationEndpoint.Type())
-	if !dstOk {
+	// Require a conversion between the wrapped type and our source
+	dstEp := destinationEndpoint.WithType(dstOpt.element)
```

Comment:
minor: `dstEp` is just an abbreviation for a variable that you already have (`destinationEndpoint`). Wondering if this would be clearer as `innerEndpoint`, `nestedEndpoint` or `elementEndpoint`? Possibly sprinkle a `destination` in there too for `destinationElementEndpoint`? Then feel free to abbreviate as appropriate if desired?

Reply:
I've made a pass through favouring longer variable names rather than contractions like `source` => `src`, `destination` => `dst` and so on.